packages/vats/src/ibc.js

```diff
@@ -74,46 +71,45 @@ export function makeIBCProtocolHandler(E, rawCallIBCDevice) {
    * @property {Counterparty} counterparty
    * @property {string} version
    *
-   * @typedef {PromiseRecord<[Endpoint, ConnectionHandler]>} OnConnectP
-   * @typedef {ConnectingInfo & { onConnectP: OnConnectP }} Outbound
+   * @typedef {PromiseRecord<AttemptDescription>} OnConnectP
+   * @typedef {Omit<ConnectingInfo, 'counterparty' | 'channelID'> & {
+   *   localAddr: Endpoint, onConnectP: OnConnectP
+   *   counterparty: { port_id: string },
+   * }} Outbound
    */
 
   /**
-   * @type {Store<string, Array<Outbound>>}
+   * @type {LegacyMap<string, Array<Outbound>>}
```

Comment:
I gather `LegacyMap` is motivated by mutability of the `Array`. Perhaps there's a sub-store arrangement that would work?

Reply:
Yes, IBC needs to be revisited once we do the SDK-wide revamp for durable/virtual collections.  I'm willing to postpone it until then.